### PR TITLE
Add flag to `donejs add` command to use defaults

### LINKED ISSUE #819: Add flag for selecting all of the default options
@phillipskevin commented on [Thu Jan 26 2017](https://github.com/donejs/cli/issues/71)

It would be nice if you could do something like

```
donejs add app my-app --yes
```

and have all of the default options selected without having to hit enter for every question.

---

@matthewp commented on [Wed Feb 15 2017](https://github.com/donejs/cli/issues/71#issuecomment-280026990)

Shouldn't this be a generator-donejs issue? You couldn't support this universally, I don't think, like `donejs add firebase --yes` wouldn't work. Hopefully the cli flags are passed to the generator some how, and they can interpret them as they wish.

---

@phillipskevin commented on [Wed Feb 15 2017](https://github.com/donejs/cli/issues/71#issuecomment-280047657)

Actually, this is a donejs/donejs issue first, and then a generator-donejs issue. We need to add a command line flag here: https://github.com/donejs/donejs/blob/master/lib/cli/index.js#L15 or commander will exit with `error: unknown option '--yes'`.

### PR DESCRIPTION
**Pull request by m-mujica**
Closes #819

### CODE REVIEW

File: lib/cli/cmd-init.js [hunk 1]
```diff
@@ -49,6 +49,10 @@ function runCliInit(folderPath, options) {
     initArgs.push('--type', options.type);
   }
 
+  if (options.yes) {
```
Comment (m-mujica) [inline lib/cli/cmd-init.js:52]: I don't like having to do this for each option we want to pass down to donejs-cli.

I looked into getting any option automatically passed down `options.options` is a list of commander [Option](http://tj.github.io/commander.js/#Option) objects. There is a private API https://github.com/tj/commander.js/blob/master/index.js#L76-L78 to get the object attribute name (the filed set at the root of the `options` object) so we could programatically check each and see if they are set, then push the the `Option.long` (the name with the `--`).

But I decided not to do this, b/c well, relying on private API didn't seem safe.

File: lib/cli/index.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ program.version(require('../../package.json').version);
 // donejs add
 program.command('add <type> [params...]')
   .option('-S, --skip-install')
+  .option('-Y, --yes', 'Use only defaults and not prompt for any option')
```
Comment (m-mujica) [inline lib/cli/index.js:16]: npm actually uses `-y` instead of `-Y`, I went with the uppercase just for the sake of consistency.

**Comment by matthewp** [comment]
I'm assuming there will be another PR in generator-donejs?

**Comment by m-mujica** [comment]
> I'm assuming there will be another PR in generator-donejs?

there you go https://github.com/donejs/generator-donejs/pull/281 
it's kind of strange that the flag being supported by `add` is actually only applying to `add app`.

**Comment by matthewp** [comment]
Once generator-donejs is done, also update donejs/cli to point to the new version of generator-donejs (wouldn't normally do this, but since it's a minor release feature I think it's a good idea).

**Pull request merged**